### PR TITLE
Fix tooltip handlers

### PR DESCRIPTION
**Pull request by cmdcolin**
This attempts to fix tooltip handlers by moving logic into the track component

I think the groundwork for this was already there and the current PileupRendering was circumventing it by trying to inject too much logic into the renderings including the tooltips

I believe by taking it out of the rendering and putting it in the "track" it will become more clear how responsibilities are located

### CODE REVIEW

**Comment by cmdcolin** [comment]
The basic API for this is more or less what I was aiming for now.

Removes Tooltip logic from PileupRendering
Removes Tooltip logic from WIggleRendering
Makes it based on showing a TooltipComponent on the BlockBasedTrack
TooltipComponent is customized for WiggleTrack and SNPCoverageTrack to be react components with custom logic and a Popper. The default TooltipComponent uses a Material-UI Tooltip

Some considerations

- Seems like there is some slowness to it. I think the slowness has always existed. The rubberband tooltip is slow. The wiggle track is slow, etc. Something is eating up CPU cycles, not sure what tho
- There is the potential for some inconsistency in styling since now there is Tooltip and Popper. Maybe the Tooltip should get converted to a popper and we invent our own style. The Tooltip is too limited, only displayed a small text snippet
- Had to put a "pointer-events: none" on the Popper because the popper causes a mouseLeave on the rendering at least when using Popper, the Tooltips didn't seem to cause this though.

File: packages/alignments/src/SNPCoverageTrack/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import Popper from '@material-ui/core/Popper'
+import Paper from '@material-ui/core/Paper'
+import { observer } from 'mobx-react'
+import { makeStyles } from '@material-ui/core/styles'
+import { Feature } from '@gmod/jbrowse-core/util/simpleFeature'
+
+const useStyles = makeStyles({
+  popper: {
+    fontSize: '0.8em',
+    zIndex: 1500, // important to have a zIndex directly on the popper itself, material-ui Tooltip uses popper and has similar thing
```
Comment (garrettjstevens) [inline packages/alignments/src/SNPCoverageTrack/Tooltip.tsx:13]:
```suggestion
    zIndex: theme.zIndex.tooltip, // important to have a zIndex directly on the popper itself, material-ui Tooltip uses popper and has similar thing
```

The theme defaults to 1500 anyway, but probably good to use the theme where possible.

**Comment by cmdcolin** [comment]
>>The Tooltip is too limited, only displayed a small text snippet

>Not sure what this means?


With tooltip we can only put a string in the "title", rather than a full component e.g. like what we want with SNPCoverage mouseovers with the table inside

File: packages/linear-genome-view/src/BasicTrack/components/BlockBasedTrack.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+})
+const Tooltip = observer(
+  (props: { model: BlockBasedTrackModel; mouseCoord: [number, number] }) => {
+    const { model, mouseCoord } = props
+    const { featureUnderMouse } = model
+    const mouseover = featureUnderMouse
+      ? getConf(model, 'mouseover', [featureUnderMouse])
+      : undefined
+    return mouseover ? (
+      <MUITooltip title={mouseover} open>
```
[the remark this answers is not in the document]
Comment (cmdcolin) [reply packages/linear-genome-view/src/BasicTrack/components/BlockBasedTrack.tsx]: right looks good to me

[the remark this answers is not in the document]
Comment (cmdcolin) [reply packages/linear-genome-view/src/BasicTrack/components/BlockBasedTrack.tsx]: could consider configurable, but maybe for now right :)

**Comment by garrettjstevens** [comment]
> With tooltip we can only put a string in the "title", rather than a full component e.g. like what we want with SNPCoverage mouseovers with the table inside

The MUI Tooltip `title` will actually take any node, so you can put any component you want in there.

**Comment by cmdcolin** [comment]
>The MUI Tooltip title will actually take any node, so you can put any component you want in there.



Interesting...maybe we should just standardize on tooltip then

**Comment by cmdcolin** [comment]
This is what the SNPCoverage tooltip could look like with and without a Paper background, and then last screenshot is plain Popper element

Screenshot from 2020-06-03 12-30-29
Screenshot from 2020-06-03 12-31-03
Screenshot from 2020-06-03 12-31-46

**Comment by garrettjstevens** [comment]
I like the MUI tooltip with no paper background best

**Comment by cmdcolin** [comment]
That seems right to me also. Then we can stylize the tooltip via themes if any changes are really needed

**Comment by peterkxie** [comment]
Making a tooltip component and taking the logic out of wiggle/snpcoverage is looking good. I like the MUI tooltip with no background (the black colored one with white text) since it fits with the other tooltips existing in Jbrowse2

**Pull request merged**